Sources/GRPCHTTP2Core/Server/HTTP2ServerTransport.swift

```diff
@@ -188,4 +188,135 @@ extension HTTP2ServerTransport.Config {
+    public struct SerializationFormat: Sendable, Equatable {
+      package enum Wrapped {
+        case pem
+        case der
+      }
+
+      package let wrapped: Wrapped
+
+      public static let pem = Self(wrapped: .pem)
+      public static let der = Self(wrapped: .der)
+    }
+
+    public struct CertificateSource: Sendable {
+      package enum Wrapped {
+        case file(path: String, format: SerializationFormat)
+        case bytes(bytes: [UInt8], format: SerializationFormat)
+      }
+
+      package let wrapped: Wrapped
+
+      /// The certificate will be provided via a file.
+      public static func file(path: String, format: SerializationFormat) -> Self {
+        Self(wrapped: .file(path: path, format: format))
+      }
+
+      /// The certificate will be provided as an array of bytes.
+      public static func bytes(_ bytes: [UInt8], format: SerializationFormat) -> Self {
+        Self(wrapped: .bytes(bytes: bytes, format: format))
+      }
+    }
+
+    public struct PrivateKeySource: Sendable {
+      package enum Wrapped {
+        case file(path: String, format: SerializationFormat)
+        case bytes(bytes: [UInt8], format: SerializationFormat)
+      }
+
+      package let wrapped: Wrapped
+
+      /// The private key will be provided via a file.
+      public static func file(path: String, format: SerializationFormat) -> Self {
+        Self(wrapped: .file(path: path, format: format))
+      }
+
+      /// The private key will be provided as an array of bytes.
+      public static func bytes(
+        _ bytes: [UInt8],
+        format: SerializationFormat
+      ) -> Self {
+        Self(wrapped: .bytes(bytes: bytes, format: format))
+      }
+    }
+
+    public struct TrustRootsSource: Sendable {
+      package enum Wrapped {
+        case certificates([CertificateSource])
+        case systemDefault
+      }
+
+      package let wrapped: Wrapped
+
+      /// Path to either a file of CA certificates in PEM format, or a directory containing CA certificates in PEM format.
+      public static func certificates(_ certificateSources: [CertificateSource]) -> Self {
+        Self(wrapped: .certificates(certificateSources))
+      }
+
+      /// The system default trust.
+      public static let systemDefault: Self = Self(wrapped: .systemDefault)
+    }
+
+    /// The certificate the server will offer during negotiation.
+    public var certificateChainSources: [CertificateSource]
+
+    /// The private key associated with the leaf certificate.
+    public var privateKeySource: PrivateKeySource
+
+    /// Whether to verify the remote certificate.
+    public var verifyClientCertificate: Bool
+
+    /// Custom trust roots to be used when verifying client certificates.
+    public var trustRoots: TrustRootsSource
+
+    /// Whether ALPN is required.
+    ///
+    /// If this is set to `true` but the client does not support ALPN, then the connection will be rejected.
+    public var requireALPN: Bool
+
+    public init(
```

Comment:
Wondering whether we should do a non-defaulting init and provide static funcs with defaults like we've done elsewhere.

I could imagine one for 'regular' TLS and one for mTLS:

```swift
static func defaults(certChainSources: [...], privateKeySource: ..., trustRoots: ...) -> Self { ... }
static func mTLS(certChainSources: [...], privateKeySource: ..., trustRoots: ...) -> Self { ... }
```

We _could_ also provide easier-to-use APIs (in the future) which take a path to the cert chain and key:

```swift
static func defaults(
  certificateChainPath: String, format: SerializationFormat, 
  privateKeyPath: String, format: SerializationFormat
) -> Self { ... }
```

Reply:
I agree with replacing the `init` with `defaults` for consistency.
Do we want to provide the `mTLS` func now though? Is the only difference when setting up the NIOSSLContext that we have to do no hostname verification for mTLS?

Reply:
I think so, yeah. I don't see any harm in adding it now tbh.